src/test/java/org/opentripplanner/middleware/tripMonitor/jobs/CheckMonitoredTripTest.java

```diff
@@ -122,36 +124,130 @@ public void canMonitorTrip() throws URISyntaxException {
         deleteMonitoredTripAndJourney(monitoredTrip);
     }
 
-    @Test
-    public void willGenerateDepartureDelayNotification() throws URISyntaxException {
-        MonitoredTrip monitoredTrip = createMonitoredTrip(user.id, otpDispatcherResponse, true);
-        OtpDispatcherResponse simulatedResponse = otpDispatcherResponse.clone();
-        Itinerary simulatedItinerary = simulatedResponse.getResponse().plan.itineraries.get(0);
-        // Set departure time to twenty minutes (in seconds). Default departure time variance threshold is 15 minutes.
-        simulatedItinerary.legs.get(0).departureDelay = 60 * 20;
+    @ParameterizedTest
+    @MethodSource("createDelayNotificationTestCases")
+    void testDelayNotifications(DelayNotificationTestCase testCase) {
+        TripMonitorNotification notification = testCase.checkMonitoredTrip.checkTripForDelay(testCase.delayType);
+        if (testCase.expectedNotificationMessage == null) {
+            assertNull(notification, testCase.message);
+        } else {
+            assertNotNull(notification, String.format("Expected notification for test case: %s", testCase.message));
+            assertEquals(testCase.expectedNotificationMessage, notification.body, testCase.message);
+        }
+    }
+
+    private static List<DelayNotificationTestCase> createDelayNotificationTestCases () throws URISyntaxException {
+        List<DelayNotificationTestCase> testCases = new ArrayList<>();
+
+        // doesn't create departure/arrival notification for on-time trip
+        CheckMonitoredTrip onTimeTrip = createCheckMonitoredTrip();
+        onTimeTrip.setJourneyState(createDefaultJourneyState());
+        testCases.add(new DelayNotificationTestCase(
+            onTimeTrip,
+            NotificationType.DEPARTURE_DELAY,
+            "doesn't create departure notification for on-time trip"
```

Comment:
For all of these test messages, is "shouldn't" a better word to use?

Reply:
Used "should" / "should not" in a8d90ab7a8ea11d1ed9c3c64e90e122e4a8a6d14